packages/flutter/lib/src/material/material_state_mixin.dart

```diff
@@ -47,6 +47,11 @@ import 'material_state.dart';
 /// }
 /// ```
 /// {@end-tool}
+///
+/// See also:
+///
+///  * [WidgetStateMixin], the generic version of `MaterialStatesController`
+///    that can be used with non-Material widgets.
 @optionalTypeArgs
 mixin MaterialStateMixin<T extends StatefulWidget> on State<T> {
```

Comment:
Can this be done with a typedef?

Reply:
Same as elsewhere, I was worried about the `debugFillProperties` returning differently.

Reply:
Same as my comment above.

Reply:
Converting this mixin into a typedef seems like a pretty big breaking change. It's methods (`setMaterialState`, `addMaterialState`, `removeMaterialState, etc.) throw errors if this is just converted into a typedef. I'm marking this as deprecated and seeing what can be helped along with a flutter fix.